### PR TITLE
Fix cgen to support union types correctly

### PR DESCRIPTION
**Pull request by aaronstaley**
See https://github.com/dropbox/dropbox-sdk-go-unofficial/pull/53

Only changes are in the python files (code generator changes). Everything else is generated code updates. (deletes are all of unused fileds)

Happy to chat in person - some knowledge of Stone (Dropbox's IDL) is needed to understand the code gen.

### CODE REVIEW

File: dropbox/async/types.go [hunk 1]
```diff
@@ -49,6 +49,10 @@ const (
 func (u *LaunchResultBase) UnmarshalJSON(body []byte) error {
 	type wrap struct {
 		dropbox.Tagged
+		// AsyncJobId : This response indicates that the processing is
+		// asynchronous. The string is an id that can be used to obtain the
+		// status of the asynchronous job.
+		AsyncJobId json.RawMessage `json:"async_job_id,omitempty"`
```
Comment (asimshankar) [inline dropbox/async/types.go]: Why make this a `json.RawMessage` and then unmarshal it again in line 65?
Could this be a `string` and then  at that point:

- Either just set `u.AsyncJobId = w.AsyncJobId=,
or
- Get rid of `wrap` altogether and `Unmarshal` directly into `u`?

I'm probably missing something.

File: dropbox/common/types.go [hunk 2]
```diff
@@ -52,6 +52,14 @@ const (
 func (u *PathRoot) UnmarshalJSON(body []byte) error {
 	type wrap struct {
 		dropbox.Tagged
+		// Root : Paths are relative to the authenticating user's root namespace
+		// (This results in `PathRootError.invalid_root` if the user's root
+		// namespace has changed.).
+		Root json.RawMessage `json:"root,omitempty"`
```
Comment (asimshankar) [inline dropbox/common/types.go]: Same comment here, I'm not following the `RawMessage` usage here.

**Comment by aaronstaley** [comment]
Alright, you inspired me to do better. :) simpler deserialization now.

File: dropbox/sdk.go [hunk 3]
```diff
@@ -37,7 +37,7 @@ const (
 	hostAPI       = "api"
 	hostContent   = "content"
 	hostNotify    = "notify"
-	sdkVersion    = "5.4.0"
+	sdkVersion    = "5.0.0"
```
Comment (asimshankar) [inline dropbox/sdk.go]: Why this change? (Seems to be downgrading a few minor versions?)

**Review by asimshankar — APPROVED** [review]
Is there some test suite?

File: dropbox/sdk.go (hunk 3 above)
Comment (aaronstaley) [reply dropbox/sdk.go]: Great eye.  I didn't provide a version argument before running the script (didn't realize the error message at the top was an error -- corrected in this diff). 

I have no idea how the numbers work (none of this is documented); I'll just choose 5.5

**Comment by aaronstaley** [comment]
Tests: other than the CI that builds, no. I added a test (linked to in the Dropbox facing PR) to verify deserialization -- sadly it depends on an updated SDK build (which I'll do next time): https://gist.github.com/aaronstaley/4230928e187414505be61bcc2b6befbb

**Review by asimshankar — APPROVED** [review]
Looks good to me. May want to set things up so that we have a branch here that tracks the upstream repo, so diffs from it are easy to spot and suggest upstream

**Comment by aaronstaley** [comment]
I'll put on my TODO to track that. Realistically, nothing is getting pushed upstream that isn't by me. :)
